libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -704,4 +736,74 @@ std::shared_ptr<arrow::DataType> make_experimental_type(const type& t) {
   return caf::visit(f, t);
 }
 
+type make_vast_type(const std::shared_ptr<arrow::DataType>& arrow_type) {
```

Comment:
clangd complains: 

> Function 'make_vast_type' is within a recursive call chain

While that is correct, it's the nature of the type system to be recursive. This doesn't happen for e.g., `make_experimental_type`, which is also defined recursively.

Reply:
You can fix this by adding a `// NOLINTNEXTLINE(misc-no-recursion)` above the function.

I think the recursion itself is fine here. Types already have an enforced maximum depth, so you cannot run into infinite recursion or a stack overflow here realistically.

I don't know why it doesn't diagnose it for `make_experimental_type` as well.